src/playdate/scoreboards.nim

```diff
@@ -21,36 +21,47 @@ import bindings/scoreboards {.all.}
 #   privateaccess(PlaydateScoreboards)
 #   playdate.scoreboards.freeScore(this.resource)
 type PDScore* = object of RootObj
-  value*: int32
-  rank*: int32
+  value*: uint32
+  rank*: uint32
   player*: string
 
 type
-  # AddScoreCallback* = proc(score: ptr PDScore, errorMessage: string)
   PersonalBestCallback* = proc(score: PDScore, errorMessage: string)
+  AddScoreCallback* = proc(score: PDScore, errorMessage: string)
   # BoardsListCallback* = proc(boards: ptr PDBoardsList, errorMessage: string)
   # ScoresCallback* = proc(scores: ptr PDScoresList, errorMessage: string)
 
 var privatePersonalBestCallback: PersonalBestCallback
+var privateAddScoreCallback: AddScoreCallback
 
 proc invokePersonalBestCallback(score: PDScorePtr, errorMessage: ConstChar) {.cdecl, raises: [].} =
   if errorMessage != nil:
     privatePersonalBestCallback(PDScore(value: 0, rank: 0, player: ""), $errorMessage)
     return
     
-  let domainScore = PDScore(value: score.value.int32, rank: score.rank.int32, player: $score.player)
+  let domainScore = PDScore(value: score.value.uint32, rank: score.rank.uint32, player: $score.player)
   playdate.scoreboards.freeScore(score)
   privatePersonalBestCallback(domainScore, $errorMessage)
 
-# proc addScore*(this: ptr PlaydateScoreboards, boardID: string, value: uint32, callback: AddScoreCallback): int32 =
-#   privateAccess(PlaydateScoreboards)
+proc invokeAddScoreCallback(score: PDScorePtr, errorMessage: ConstChar) {.cdecl, raises: [].} =
+  if errorMessage != nil:
+    privateAddScoreCallback(PDScore(value: 0, rank: 0, player: ""), $errorMessage)
+    return
+    
+  let domainScore = PDScore(value: score.value.uint32, rank: score.rank.uint32, player: $score.player)
+  playdate.scoreboards.freeScore(score)
```

Comment:
Note I took a bit of a different approach here: instead of keeping a PDScoreObj which has a resource to be freed when it is destroyed; I'm copying the data to a Nim Object which i epect to be memory managed. We can then immediately free the PDScorePtr

Reply:
Why the difference?

Reply:
I was trying a bunch of things because my code was crashing. It turns out that my code was okay, but there is either a bug or incompatibility between the 2.6.0 api and the currently published 2.5.0 docs.

Now that I know this I can probably rewrite  it in the way that is established in the rest of the codebase.

I personally like to detach the domain model from the api model. It allows to smooth over data types that aren't directly compatible between nim and c by explicitly mapping from one to the other. Part of the reason might also be that I'm still a bit scared of memory management and this way it is all nicely contained within the callback. 

But yeah, I can see the benefit of a consistent approach. I presume you prefer that?

Reply:
I'm fine with leaving these as nim objects. My only concern might be memory allocations -- but I think this will be invoked few enough times in a process that it isn't an issue.